numpy/_core/tests/test_strings.py

```diff
@@ -796,6 +796,54 @@ def test_zfill(self, buf, width, res, dt):
         assert_array_equal(np.strings.zfill(buf, width), res)
 
 
+@pytest.mark.parametrize("dt", [
+    "U",
+    "S",
+    pytest.param("T", marks=pytest.mark.xfail(
+        reason="StringDType support not implemented",
+        strict=True),
+    ),
+])
+class TestMethodsWithoutStringDTypeSupport:
+    @pytest.mark.parametrize("buf,sep,res1,res2,res3", [
+        ("this is the partition method", "ti", "this is the par",
+            "ti", "tion method"),
+        ("http://www.python.org", "://", "http", "://", "www.python.org"),
+        ("http://www.python.org", "?", "http://www.python.org", "", ""),
+        ("http://www.python.org", "http://", "", "http://", "www.python.org"),
+        ("http://www.python.org", "org", "http://www.python.", "org", ""),
+        ("mississippi", "ss", "mi", "ss", "issippi"),
+        ("mississippi", "i", "m", "i", "ssissippi"),
+        ("mississippi", "w", "mississippi", "", ""),
+    ])
+    def test_partition(self, buf, sep, res1, res2, res3, dt):
+        buf = np.array(buf, dtype=dt)
+        sep = np.array(sep, dtype=dt)
+        res1 = np.array(res1, dtype=dt)
+        res2 = np.array(res2, dtype=dt)
+        res3 = np.array(res3, dtype=dt)
+        assert_array_equal(np.strings.partition(buf, sep), (res1, res2, res3))
```

Comment:
The tuple does an implicit conversion to a new array. I'd write 
```
p1, p2, p3 = np.strings.partition(buf, sep)
assert_array_equal(p1, res1)
assert_array_equal(p2, res2)
assert_array_equal(p3, res3)
assert_array_equal(p1 + p2 + p3, buf)
```
Actually not 100% sure the latter works...